arcadexyz/arcade-protocol
# Scripts

This PR updates the deployment, role setting and contract verification scripts for V3 and adds a script for whitelisting the payable currencies and collateral currently verified on the Dapp.

The contracts are deployed and verified on `Sepolia` testnet.

To run the e2e script:
`EXEC=true yarn test-deploy --network sepolia`

To test the whitelisting script, 
`npx hardhat run scripts/token-whitelist.ts --network sepolia` :
You would first need to comment out lines 151 - 157 in `token-whitelist.ts` so that the deployer can still run whitelisting on origination controller before they renounce their `WHITELIST_MANAGER_ROLE` role. 
Next, deploy a new set of contracts and make sure the `originationControllerFactory.attach()` is referencing the new ogc deployment. 
Sometimes you have to run the script twice for the whitelisting of the payable currencies to take (it is after the collateral whitelisting).
After the whitelisting of all tokens is complete, uncomment lines 151 - 157 and have the deployer renounce their `WHITELIST_MANAGER_ROLE` role. 

Sepolia faucets:
https://faucet.sepolia.dev/
https://sepoliafaucet.com/
https://faucet.quicknode.com/ethereum/sepolia
https://sepolia-faucet.pk910.de/
